source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -73,13 +73,27 @@ struct SaveRestoreContext {
 #define CALL_WITH_CONTEXT(_x, _context)                                                            \
   do {                                                                                             \
     SaveRestoreContext _saved_context(static_cast<Context*>(_context));                            \
-    _x;                                                                                            \
+    WAVM::Runtime::catchRuntimeExceptions([&] {                                                     \
+      _x;                                                                                          \
+    }, [&](WAVM::Runtime::Exception* exception) {                                              \
+      auto description = describeException(exception);                                            \
+      delete exception; \
```

Comment:
While this might be equivalent at this time, I think you should call `destroyException(exception)` here.

Reply:
done.